File: app/src/main/java/com/netvirta/netvision/OpencvNativeClass.java
package com.netvirta.netvision;

public class OpencvNativeClass {

    static {
        System.loadLibrary("my-opencv-lib");
    }

    public native static int convertGray(long matAddrRgba, long matAddrGray);

    public native static void faceDetection(long rgbaAddress);

    public native static void bodyDetection(long rgbaAddress);
}

File: app/src/main/java/com/netvirta/netvision/OpencvNativeClass.kt
package com.netvirta.netvision

object OpencvNativeClass {

    init {
        System.loadLibrary("my-opencv-lib")
    }

    external fun convertGray(matAddrRgba: Long, matAddrGray: Long): Int

    external fun faceDetection(rgbaAddress: Long)

    external fun bodyDetection(rgbaAddress: Long)

    external fun lineDetection(rgbaAddress: Long)
}
